lib/int64.js

```diff
@@ -0,0 +1,377 @@
+      this.value[0] = -numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(-numValue / (UINT32_MAX + 1));
+      this.value = Int64._conversion(this.value);
+    } else if (typeof value === 'string') {
+      const negative = value.startsWith('-');
+      if (negative) value = value.slice(1);
+
+      let radix = 10;
+      if (value.startsWith('0x')) {
+        radix = 16;
+      } else if (value.startsWith('0o')) {
+        radix = 8;
+      } else if (value.startsWith('0b')) {
+        radix = 2;
+      }
+      if (radix !== 10) value = value.slice(2);
+      const intRadix = new Int64(radix);
+      let res = new Int64();
+      for (let i = 0; i < value.length; i++) {
+        const digit = charToNum[value[i]];
+        const intDigit = new Int64(digit);
+        res = Int64.mult(res, intRadix);
+        if (negative) res.sub(intDigit);
+        else res.add(intDigit);
+      }
+      this.value[0] = res.value[0];
+      this.value[1] = res.value[1];
+    }
+  }
+
+  toInt32() {
+    const mask = ~(1 << 31);
+    if (this.value[1] >>> 31 === 1) {
+      const value = Int64._conversion(this.value);
+      return -(value[0] & mask);
+    } else {
+      return this.value[0] & mask;
+    }
+  }
+
+  // Convert signed to 2's complement representation and vise versa
+  static _conversion(value) {
+    const result = new Uint32Array(value);
+    let pos = 0;
+    if (result[0] === 0) {
+      while (pos < 32 && ((result[1] >>> pos) & 1) === 0) {
+        pos++;
+      }
+      if (pos !== 31) {
+        const mask = (1 << (pos + 1)) - 1;
+        result[1] = (~(result[1] | mask)) | (result[1] & mask);
+      }
+    } else {
+      while (pos < 32 && ((result[0] >>> pos) & 1) === 0) {
+        pos++;
+      }
+      if (pos !== 31) {
+        const mask = (1 << (pos + 1)) - 1;
+        result[0] = (~(result[0] | mask)) | (result[0] & mask);
+      }
+      result[1] = ~result[1];
+    }
+    return result;
+  }
+
+  add(b) {
+    if (b.value[0] === 0 && b.value[1] === 0) return this;
+    const tmp = this.value[0] + b.value[0];
+    this.value[0] = tmp;
+    this.value[1] += b.value[1] + (tmp > UINT32_MAX ? 1 : 0);
+    return this;
+  }
+
+  static add(a, b) {
+    return new Int64(a).add(b);
+  }
+
+  sub(b) {
+    if (b.value[0] === 0 && b.value[1] === 0) return this;
+    const value = Int64._conversion(b.value);
+    const tmp = this.value[0] + value[0];
+    this.value[0] = tmp;
+    this.value[1] += value[1] + (tmp > UINT32_MAX ? 1 : 0);
+    return this;
+  }
+
+  static sub(a, b) {
+    return new Int64(a).sub(b);
+  }
+
+  static cmp(a, b) {
+    const aSign = a.value[1] >>> 31;
+    const bSign = b.value[1] >>> 31;
+    if (aSign !== bSign) return aSign === 1 ? -1 : 1;
+
+    if (a.value[1] > b.value[1]) return 1;
+    if (a.value[1] < b.value[1]) return -1;
+    if (a.value[0] === b.value[0]) return 0;
+    return a.value[0] > b.value[0] ? 1 : -1;
+  }
+
+  static _division(n, d) {
+    const zero = new Int64();
```

Comment:
I'd prefer to have something like
```js
Int64.getZero() {
  return new Int64(0);
}
```
otherwise, every operation that modifies value should be done like this
```js
const zero = new Int64(Int64.zero);
```
so the value won't be overwritten.

Reply:
+1, but maybe just `Int64.zero()`?